app/src/main/java/de/danoeh/antennapod/fragment/DiscoveryFragment.java

```diff
@@ -97,12 +120,81 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
             intent.putExtra(OnlineFeedViewActivity.ARG_FEEDURL, podcast.feedUrl);
             startActivity(intent);
         });
+
+        List<String> countryCodeArray = new ArrayList<String>(Arrays.asList(Locale.getISOCountries()));
+        HashMap<String, String> countryCodeNames = new HashMap<String, String>();
+        for (String code: countryCodeArray) {
+            Locale locale = new Locale("", code);
+            String countryName = locale.getDisplayCountry();
+            if (countryName != null) {
+                countryCodeNames.put(code, countryName);
+            }
+        }
+        //countryCodeArray.add(0, getResources().getString(R.string.discover_hide_fake_code));
+        // countryCodeNames.put(getResources().getString(R.string.discover_hide_fake_code),
+        //                 getResources().getString(R.string.discover_hide));
+
+        List<String> countryNamesSort = new ArrayList<String>(countryCodeNames.values());
+        Collections.sort(countryNamesSort);
+        countryNamesSort.add(0, getResources().getString(R.string.discover_hide));
+
+        Spinner countrySpinner = root.findViewById(R.id.spinner_country);
+        ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(this.getContext(), 
+                android.R.layout.simple_spinner_item, 
+                countryNamesSort);
+        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        countrySpinner.setAdapter(dataAdapter);
+        int pos = countryNamesSort.indexOf(countryCodeNames.get(countryCode));
+        countrySpinner.setSelection(pos);
+
+        countrySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> countrySpinner, View view, int position, long id) {
+                String countryName = (String) countrySpinner.getItemAtPosition(position);
+
+                if (countryName.equals(getResources().getString(R.string.discover_hide))) {
+                    countryCode = getResources().getString(R.string.discover_hide_fake_code);
+                    gridView.setVisibility(View.GONE);
```

Comment:
Hiding like this only works directly after changing the setting, right? What about moving the hiding code to `loadToplist`?

Reply:
done